src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.ts

```diff
@@ -34,21 +34,22 @@ export class EnrichedComplex {
   styleUrls: ['./table-interactor-column.component.css']
 })
 export class TableInteractorColumnComponent implements OnChanges {
-  @Input() complexes: Element[];
-  @Input() interactorsSorting: string;
-  @Input() interactors: Interactor[];
-  @Input() organismIconDisplay: boolean;
-  @Input() interactorTypeDisplay: boolean;
-  @Input() IDDisplay: boolean;
-
+  complexes = input<Element[]>([]);
+  interactorsSorting = input<string>();
+  interactors = input<Interactor[]>([]);
+  organismIconDisplay = input<boolean>(true);
+  interactorTypeDisplay = input<boolean>(true);
+  idDisplay = input<boolean>(true);
+
+  // TODO rework bellow to compute when needed from the inputs
```

Comment:
This already happens, right? We only call `enrichInteractors` in ngOnChanges if there are changes in interactors input.

Reply:
Yes in principle indeed, but ngOnChanges is called for any changes, not specifically for the concerned inputs.
Using signals would be quite powerful here, but as stated in my comments, not so straightforward  as the code is kinda convoluted over this component.

Reply:
That's why there's an if inside ngOnChanges. ngOnChanges is called for any changes, but it has an argument with the changes. So we check if the field `interactors` is part of the changes to call `enrichInteractors`.

There may cleaner ways to do it with the Angular upgrade, but we're not currently calling `enrichInteractors` every time `ngOnChanges` is called.

Reply:
Yes yes of course, just saying it would be a really nice place to use the new signals, but indeed it is not required